### PR TITLE
FEM-10479 initial udp playback support

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java [hunk 1]
```diff
@@ -396,10 +399,10 @@ private MediaSource buildInternalExoMediaSource(PKMediaSourceConfig sourceConfig
             // mp4 and mp3 both use ExtractorMediaSource
             case mp4:
             case mp3:
+            case udp:
                 mediaSource = new ProgressiveMediaSource.Factory(dataSourceFactory)
```
Comment (OrenMe) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java:411]: Seems we need to use ExtractorMediaSource
https://medium.com/@jatinmishra27/playing-udp-stream-in-exoplayer2-55ea3ea6e1db

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java [hunk 2]
```diff
@@ -273,6 +273,10 @@ private DefaultTrackSelector initializeTrackSelector() {
     }
 
     private void preparePlayer(@NonNull PKMediaSourceConfig sourceConfig) {
+        if (this.sourceConfig != null && this.sourceConfig.mediaSource.getMediaFormat() == PKMediaFormat.udp) {
+            player.seekTo(0);
```
Comment (OrenMe) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: this is for seek to live edge?

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java [hunk 3]
```diff
@@ -399,10 +405,16 @@ private MediaSource buildInternalExoMediaSource(PKMediaSourceConfig sourceConfig
             // mp4 and mp3 both use ExtractorMediaSource
             case mp4:
             case mp3:
-            case udp:
+            //case udp:
```
Comment (OrenMe) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: if not required please remove

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java (hunk 2 above)
Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: no the r seek to live edge is done automatically once media is defined as live

I have updated the logic so now it is more readable and achieve the same goal that playback after change media from udp to hls/dash will make player to return the correct position

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java [hunk 4]
```diff
@@ -399,7 +409,15 @@ private MediaSource buildInternalExoMediaSource(PKMediaSourceConfig sourceConfig
                 mediaSource = new ProgressiveMediaSource.Factory(dataSourceFactory)
                         .createMediaSource(uri);
                 break;
-
+            case udp:
+                DataSource.Factory udpDatasourceFactory = () -> new UdpDataSource(3000, 100000);
```
Comment (OrenMe) [inline playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: are these values for `maxPacketSize` and `socketTimeoutMillis` the preferred values?
Maybe we should expose as settings.

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java (hunk 3 above)
Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: done

File: playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java (hunk 4 above)
Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: will check this

Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: @OrenMe  done

Comment (giladna) [reply playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java]: @noamtm  please review

**Review by OrenMe — APPROVED** [review]

**Pull request merged**